_posts/2022-01-28-filesystem-encryption-integrity.adoc

```diff
@@ -0,0 +1,145 @@
+---
+layout: post
+title: "Encrypting Filesystem realms and enabling integrity"
+date: 2022-01-28
+tags: filesystem-realm encryption integrity
+synopsis: An overview of the new filesystem realm encryption and integrity in the Elytron Security Realms.
+author: araskar
+link:
+---
+
+Previously, any data in the filesystem realm was always stored unencrypted and data was not verified. This meant that anyone with access to the filesystem identity files could access the credentials, and attributes (such as assigned roles). Along with that if anyone were to change the data in file, the realm would have no way of knowing. Now both encryption has been added via a SecretKey, and filesystem integrity has been added by generating a signature via a Public and Private Keypair.
+
+== An overview of the new attributes
+
+The filesystem realm now supports the following attributes
+
+* ``credential-store``: This attribute specifies the credential store resource where the secret key resides in. This credential store must be configured prior to creating the filesystem realm. This attribute is optional and if not specified, the filesystem realm will not be encrypted.
+
+* ``secret-key``: This attribute specifies the alias to the secret key in the credential store to encrypt the relam. By default the secret key alias is set to `key` when creating the credential store. This attribute is required if the `credential-store` attribute is specified.
+
+* ``key-store``: This attribute specifies the key store resource where the key pair resides in. This key store must be configured prior to creating the filesystem realm. This attribute is optional and if not specified, the filesystem realm will not verify it's integrity.
+
+* ``key-store-alias``: This attribute specifies the alias to the key pair in the key store to verify filesystem integrity. This attribute is required if the `key-store` attribute is specified.
+
+== Reviewing the Elytron Configuration
```

Comment:
Might be good to rename this section (maybe something like "A Complete Example" or something along those lines). It would also be good to give an overview of what you're about to explain. This post might give some ideas on how to structure this next part:

https://wildfly-security.github.io/wildfly-elytron/blog/multi-tenancy-support-openid-connect/